src/Tools/dotnet-monitor/ParameterCapturing/CaptureParametersOperation.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring;
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tools.Monitor.HostingStartup;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Drawing.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.ParameterCapturing
+{
+    internal sealed class CaptureParametersOperation : IInProcessOperation
+    {
+        private readonly ProfilerChannel _profilerChannel;
+        private readonly IEndpointInfo _endpointInfo;
+        private readonly ILogger _logger;
+        private readonly MethodDescription[] _methods;
+        private readonly TimeSpan _duration;
+
+        public CaptureParametersOperation(IEndpointInfo endpointInfo, ProfilerChannel profilerChannel, ILogger logger, MethodDescription[] methods, TimeSpan duration)
+        {
+            _profilerChannel = profilerChannel;
+            _endpointInfo = endpointInfo;
+            _logger = logger;
+            _methods = methods;
+            _duration = duration;
+        }
+
+        private async Task<bool> IsEndpointIsCapableOfProcessingRequestsAsync(CancellationToken token)
```

Comment:
nit: Probably either `IsEndpointCapableOfProcessingRequestsAsync` or `EndpointIsCapableOfProcessingRequestsAsync` (double `is`)